ql/src/java/org/apache/hadoop/hive/ql/ddl/view/materialized/alter/rebuild/NonNativeAcidMaterializedViewASTBuilder.java

```diff
@@ -39,6 +44,28 @@ public List<ASTNode> createDeleteSelectNodes(String tableName) {
             .collect(Collectors.toList()));
   }
 
+  @Override
+  public void appendDeleteSelectNodes(ASTNode selectNode, String tableName) {
+    Set<String> selectedColumns = new HashSet<>(selectNode.getChildCount());
+
+    for (int i = 0; i < selectNode.getChildCount(); ++i) {
+      ASTNode selectExpr = (ASTNode) selectNode.getChild(i);
+      ASTNode expression = (ASTNode) selectExpr.getChild(0);
```

Comment:
could we get an IndexOutOfBounds exception here?

Reply:
Select expression (TOK_SELEXPR) has at least one child and it is the expression we looking for here.
https://github.com/apache/hive/blob/5e78ce0e1f1577c4b68f9149430c1d6ad26fd91d/parser/src/java/org/apache/hadoop/hive/ql/parse/SelectClauseParser.g#L80-L89